crypto/evp/evp_enc.c

```diff
@@ -174,9 +177,11 @@ int EVP_CipherInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
             /* fall-through */
 
         case EVP_CIPH_CBC_MODE:
-
-            OPENSSL_assert(EVP_CIPHER_CTX_iv_length(ctx) <=
-                           (int)sizeof(ctx->iv));
+            if (!ossl_assert(EVP_CIPHER_CTX_iv_length(ctx) <=
+                             (int)sizeof(ctx->iv))) {
+                EVPerr(EVP_F_EVP_CIPHERINIT_EX, ERR_R_INTERNAL_ERROR);
```

Comment:
Same argument as above. With additional twist. `default:` probably deserves corresponding treatment. Just to be clear, I for one for plain assert. In such case there should be plain assert in `default:` clause. But if one chooses to pass error, then `default:` should do same.